sideload.go

```diff
@@ -0,0 +1,240 @@
+package kapacitor
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strconv"
+	"text/template"
+	text "text/template"
+
+	"github.com/influxdata/kapacitor/bufpool"
+	"github.com/influxdata/kapacitor/edge"
+	"github.com/influxdata/kapacitor/keyvalue"
+	"github.com/influxdata/kapacitor/models"
+	"github.com/influxdata/kapacitor/pipeline"
+	"github.com/influxdata/kapacitor/services/sideload"
+	"github.com/pkg/errors"
+)
+
+type SideloadNode struct {
+	node
+	s          *pipeline.SideloadNode
+	source     sideload.Source
+	orderTmpls []orderTmpl
+
+	order []string
+
+	bufferPool *bufpool.Pool
+}
+
+// Create a new  SideloadNode which shifts points and batches in time.
+func newSideloadNode(et *ExecutingTask, n *pipeline.SideloadNode, d NodeDiagnostic) (*SideloadNode, error) {
+	sn := &SideloadNode{
+		node:       node{Node: n, et: et, diag: d},
+		s:          n,
+		bufferPool: bufpool.New(),
+		order:      make([]string, len(n.OrderList)),
+		orderTmpls: make([]orderTmpl, len(n.OrderList)),
+	}
+	u, err := url.Parse(n.Source)
```

Comment:
Should lines 40-49 go into the et.tm.SideloadService.Source method?

Reply:
Yeah, that would make more sense.